src/main/java/org/apache/brooklyn/cloudfoundry/utils/LocalResourcesDownloader.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.cloudfoundry.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.brooklyn.util.core.ResourceUtils;
+import org.apache.brooklyn.util.os.Os;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.commons.io.FileUtils;
+
+import com.google.common.base.Throwables;
+
+public class LocalResourcesDownloader {
```

Comment:
Applications artifacts has to be updated from a local path. 
So, when a online artifact is used for example `path: https://github.com/kiuby88/brooklyn-cloudfoundry/raw/feature/vanilla-cf-application-loc-management/src/test/resources/brooklyn-example-hello-world-sql-webapp-in-paas.war` this downloader is used to download the artifact, save it into `tmp` folder and get the local (canonical) path of the resource.

Then, when the artifacts are pushed from local path according to the next pattern:
```
File war = LocalResourcesDownloader.downloadResourceInLocalDir(uri);
path = war.getCanonicalPath();
````